addon/components/polaris-resource-list/bulk-actions.js

```diff
@@ -0,0 +1,535 @@
+   *
+   * @type {Function}
+   * @default noop
+   * @property onSelectModeToggle
+   */
+  onSelectModeToggle(/* selectMode **/) {},
+
+  /**
+   * Callback when the select all checkbox is clicked
+   *
+   * @type {Function}
+   * @default noop
+   * @property onToggleAll
+   */
+  onToggleAll() {},
+
+  instanceOfBulkActionListSectionArray(actionsCollection) {
+    let validList = actionsCollection.filter((action) => {
+      return action.items;
+    });
+
+    return actionsCollection.length === validList.length;
+  },
+
+  instanceOfBulkActionArray(actionsCollection) {
+    let validList = actionsCollection.filter((action) => {
+      return !action.items;
+    });
+
+    return actionsCollection.length === validList.length;
+  },
+
+  setContainerNode() {
+    this.set('containerNode', this.element);
+  },
+
+  setMoreActionsNode(node) {
+    this.set('moreActionsNode', node);
+  },
+
+  setLargeScreenButtonsNode(node) {
+    this.set('largeScreenButtonsNode', node);
+  },
+
+  handleResize() {
+    throttle(
+      this,
+      function() {
+        let {
+          smallScreenPopoverVisible,
+          largeScreenPopoverVisible,
+          containerNode,
+        } = this.getProperties(
+          'smallScreenPopoverVisible',
+          'largeScreenPopoverVisible',
+          'containerNode'
+        );
+
+        if (containerNode) {
+          let containerWidth = containerNode.getBoundingClientRect().width;
+          if (containerWidth > 0) {
+            this.set('containerWidth', containerWidth);
+          }
+        }
+
+        if (smallScreenPopoverVisible || largeScreenPopoverVisible) {
+          this.setProperties({
+            smallScreenPopoverVisible: false,
+            largeScreenPopoverVisible: false,
+          });
+        }
+      },
+      50
+    );
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    let promotedActions = this.get('promotedActions');
+
+    if (promotedActions && promotedActions.length > MAX_PROMOTED_ACTIONS) {
+      warn(
+        `To provide a better user experience. There should only be a maximum of ${MAX_PROMOTED_ACTIONS} promoted actions.`
+      );
+    }
+
+    this.setContainerNode();
+
+    let largeScreenGroupNode = this.element.querySelector(
+      '.Polaris-ResourceList-BulkActions__Group--largeScreen'
+    );
+
+    if (largeScreenGroupNode) {
+      this.addEventListener(largeScreenGroupNode, 'resize', this.handleResize);
+    }
+  },
+
+  didRender() {
+    this._super(...arguments);
+
+    let moreActionsEl = this.element.querySelector(
+      '.Polaris-ResourceList-BulkActions__Popover'
+    );
+
+    let largeScreenButtonsEl = this.element.querySelector(
+      '.Polaris-ResourceList-BulkActions__Group--largeScreen'
+    );
+
+    if (moreActionsEl) {
+      this.setMoreActionsNode(moreActionsEl);
+    }
+
+    if (largeScreenButtonsEl) {
+      this.setLargeScreenButtonsNode(largeScreenButtonsEl);
+    }
```

Comment:
This seems like a bit of a clunky approach to imitating React's `ref` behaviour - in the past we've handled this by defining [CPs](https://github.com/smile-io/ember-polaris/blob/master/addon/mixins/components/svg-handling.js#L50) that look up the element, do you think that approach would be cleaner here?